### PR TITLE
Add objects printer

### PR DESCRIPTION
**Pull request by pieterv**
This covers most of the object printing cases. Im pretty sure i covered all the weird edge cases around `set`, `async`, computed properties etc.

### CODE REVIEW

File: src/__tests__/__snapshots__/fjs_babel-test.js.snap [hunk 1]
```diff
@@ -82,8 +81,7 @@ if (foo) {
   foobar();
 }
 
-if (foo) {
-}
+if (foo) {}
```
Comment (pieterv) [inline src/__tests__/__snapshots__/fjs_babel-test.js.snap]: Maybe we want to treat empty if blocks differently to normal blocks but for now it seems like a nicer default to collapse these.

File: src/__tests__/__snapshots__/fjs_babel-test.js.snap [hunk 2]
```diff
@@ -137,6 +135,27 @@ foo[test()][bar()];
...
+"var foo = {};
+
+var foo = { x, y };
+
+var foo = { x: x, y: y };
+
+var foo = { x: x, y: y };
+
+var foo = {
+  [\'bar\']: \'foo\',
```
Comment (pieterv) [inline src/__tests__/__snapshots__/fjs_babel-test.js.snap]: I dont understand how it decides to escape things :/

But the good news is we have real computed properties now :D

File: src/__tests__/__snapshots__/fjs_babel-test.js.snap [hunk 3]
```diff
@@ -201,13 +220,13 @@ exports[`test XJSSpreadAttribute.js 1`] = `
...
 1 + a = 2;
-a = a || a = {  };
+a = a || a = {};
 "
 `;
 
 exports[`test assignment.js 1`] = `
 "x = 1;
-var { y: y = 1 } = obj;
+var { y = 1 } = obj;
```
Comment (pieterv) [inline src/__tests__/__snapshots__/fjs_babel-test.js.snap]: :D i didnt even realise you could do this until i saw this test.

File: src/__tests__/__snapshots__/fjs_babel-test.js.snap [hunk 4]
```diff
@@ -359,10 +362,10 @@ exports[`test computed-property-comments-2.js 1`] = `
 "var test = { /**
      * Before bracket init
      */
-\'a\': \'1\', /*
+[\'a\']: \'1\', [/*
```
Comment (pieterv) [inline src/__tests__/__snapshots__/fjs_babel-test.js.snap]: This should probably have a break here. Can fix it later though

File: src/__tests__/__snapshots__/fjs_babel-test.js.snap [hunk 5]
```diff
@@ -457,6 +459,20 @@ exports[`test if-line-comment.js 1`] = `
 "
 `;
 
+exports[`test jsx.js 1`] = `
+"var single = \'quotes\';
+var outnumber = \'double\';
+var moreSingleQuotesThanDouble = \'!\';
+
+React.createClass({ render() {
```
Comment (pieterv) [inline src/__tests__/__snapshots__/fjs_babel-test.js.snap]: We should probably break the full block if the inner content has some kinda break in it..?

Comment (kyldvs) [inline src/__tests__/__snapshots__/fjs_babel-test.js.snap]: Yeah breaking scopes if any child scope breaks is something I need to add into the rules. There are quite a few other places where this makes sense to do.

File: src/printers/object.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+'use strict';
+
+import Tokens from '../Tokens';
+
+import {printList} from '../utils';
+
+export default {
+  ObjectProperty: ({print, node}) => [
+    node.computed
+      ? [
```
Comment (kyldvs) [inline src/printers/object.js]: I can't wait to make more progress and run the formatter on its own source to see what it does :P
